public/app/features/dashboard-scene/scene/ScopeSelectorScene.tsx

```diff
@@ -0,0 +1,238 @@
+import { css, cx } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2, Scope, ScopeDashboard, SelectableValue } from '@grafana/data';
+import {
+  SceneComponentProps,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneObjectUrlSyncConfig,
+  SceneObjectUrlValues,
+} from '@grafana/scenes';
+import { IconButton, Select, useStyles2 } from '@grafana/ui';
+
+export interface ScopeSelectorSceneState extends SceneObjectState {
+  isExpanded: boolean;
+
+  scopes: Scope[];
+  isScopesLoading: boolean;
+
+  pendingValue: string | undefined;
+  value: string | undefined;
+
+  dashboards: ScopeDashboard[];
+  isDashboardsLoading: boolean;
```

Comment:
My comment about this in the design doc was about sparating the ScopeSelector object into two:

- ScopeSelector
  - ScopeSelectorDashbaordLists

The general idea would be to have the list as a re-usable chunk, rather than a piece of code hardcoded in to sche ScopeSelector. The list would just subscribe to the ScopeSelector state and refresh the list of dashbaords accordingly. This way we literally have two dummy fetch-only objects that play well together.

Reply:
Done!